stock_move_backdating/models/stock_picking.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Alex Comba - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.multi
+    def do_transfer(self):
+        result = super(StockPicking, self).do_transfer()
+        for picking in self:
+            # set date_done as the youngest date among the moves
+            dates = picking.mapped('move_lines.date')
+            picking.write({'date_done': max(dates)})
+            for move in picking.move_lines:
+                # set date on account.move same as date on stock.move
+                account_moves = self.env['account.move'].search(

```

Comment:
@tafaRU Please avoid searching in a loop.

Get all account moves before loop.

Reply:
could you please show me how to modify it accordingly?

Reply:
```python
all_moves = self.env['account.move'].search([('ref', '=', picking.name), ('line_ids.name', 'in', picking.mapped('move_lines.name'))])
for move in picking.move_lines:
    for account_move in all_moves.filtered(lambda m, pm=move: pm.name in m.line_ids.mapped('name')):
        account_move.date = move.date
```
Don't forget every search is a read in DB, here you fetch all records once and then, just python code

Reply:
@tafaRU The only thing I oppose to this kind of logic is that you rely on name to map two models, is it accurate (can't we have false values) ?

Reply:
The `account.move.line` values are prepared at https://github.com/odoo/odoo/blob/10.0/addons/stock_account/models/stock.py#L274 so the `name` should not have false values in respect of https://github.com/odoo/odoo/blob/10.0/addons/stock/models/stock_move.py#L26

Reply:
@tafaRU Yes I understand. What I said is relying on name and not ids to map two models is not accurate. But you don't have other means to do so.